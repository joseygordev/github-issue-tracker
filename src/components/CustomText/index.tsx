import React from 'react';
import { Text as DefaultText } from 'react-native';

import { useThemeColor } from '../../hooks/useThemeColor';

type ThemeProps = {
  lightColor?: string;
  darkColor?: string;
};

export type TextProps = ThemeProps & DefaultText['props'];

export default function CustomText(props: TextProps) {
  const { style, lightColor, darkColor, ...otherProps } = props;
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  return <DefaultText style={[{ color }, style, { fontFamily: 'space-mono' }]} {...otherProps} />;
}