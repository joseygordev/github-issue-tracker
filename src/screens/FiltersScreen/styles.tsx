import { StyleSheet } from 'react-native';
import Colors from '../../constants/Colors';

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    padding: 16,
    justifyContent: 'space-between',
  },
  filter: {
    borderRadius: 16,
    paddingHorizontal: 16,
    paddingVertical: 4,
    paddingTop: 6,
    marginVertical: 8,
    borderWidth: 1,
    flexDirection: 'row',
    alignItems: 'center',
    height: 42,
  },
  activeFilter: {
    borderColor: Colors.blue,
  },
  inputs: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  button: {
    alignSelf: 'center',
    minWidth: '100%'
  }
});
