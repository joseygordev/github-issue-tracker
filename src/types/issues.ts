import { Comment } from './comments';
import { RootTabScreenProps } from './navigation';

export interface Issue {
  url: string;
  id: number;
  number: string;
  title: string;
  body: string;
  user: IssueUser;
  labels: IssueLabel[];
  created_at: string;
  state: 'open' | 'closed';
  comments_url: string;
  pull_request?: object;
  isBookmarked?: boolean;
}

export interface IssueUser {
  login: string;
  id: number;
  avatar_url: string;
  url: string;
  html_url: string;
}

export interface IssueLabel {
  id: string;
  url: string;
  name: string;
  color: string;
  default: boolean;
  description: string;
}

export interface IssueFilter {
  id: string;
  label: string;
  isActive: boolean;
}

export interface IssueState {
  issues: Issue[];
  loading: boolean;
  error?: Error;
  owner: string;
  repo: string;
  filters: IssueFilter[];
  currentPage: number;
}

export interface IssueAction {
  type: IssueActionType;
  payload?: Issue[] | Comment[] | string | number | Error | string[];
}

export interface GetIssuesParams {
  owner: string;
  repo: string;
  filter: string;
  page: number;
}

export interface IssueItemProps {
  issue: Issue;
  navigation: RootTabScreenProps<'Issues'|'Bookmark'>;
}

export interface IssueHeaderProps {
  issue: Issue;
}

export enum IssueActionType {
  SET_ISSUES_LOADING = 'SET_ISSUES_LOADING',
  SET_ISSUES_SUCCESS = 'SET_ISSUES_SUCCESS',
  SET_ISSUES_ERROR = 'SET_ISSUES_ERROR',
  COMMIT_OWNER = 'COMMIT_OWNER',
  COMMIT_REPO = 'COMMIT_REPO',
  TOGGLE_FILTER = 'TOGGLE_FILTER',
  SET_PAGE = 'SET_PAGE',
}

export interface useIssuesReturnedValue {
  data: {
    issues: Issue[];
    loading: boolean;
    filters: IssueFilter[];
    filtersActived: IssueFilter[];
    currentPage: number;
    error?: Error;
  };
  actions: {
    getIssues: () => void;
    setFilter: (filterIds: string[]) => void;
    setPage: (page: number) => void;
  };
}
export interface useCurrentScreenReturnValue {
  data: {
    owner: string;
    repo: string;
    error?: Error;
  };
  actions: {
    commitOwner: (id: string) => void;
    commitRepo: (id: string) => void;
  };
}
