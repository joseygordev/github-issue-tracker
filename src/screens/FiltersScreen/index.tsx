import React, { useState } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { AntDesign } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';

import useIssues from '../../hooks/useIssues';
import useIssuesScreen from '../../hooks/useIssuesScreen';

import Colors from '../../constants/Colors';

import CustomView from '../../components/CustomView';
import CustomText from '../../components/CustomText';
import CustomTextInput from '../../components/CustomTextInput';
import CustomButton from '../../components/CustomButton';

import styles from './styles';
import { useEffect } from 'react';

export default function FiltersScreen() {
  const navigation = useNavigation();
  const issuesManager = useIssues();
  const { filters } = issuesManager.data;
  const { setFilter } = issuesManager.actions;

  const currentPageManager = useIssuesScreen();
  const { owner: commitedOwner, repo: commitedRepo } = currentPageManager.data;
  const { commitOwner, commitRepo } = currentPageManager.actions;

  const [owner, setOwner] = useState(commitedOwner);
  const [repo, setRepo] = useState(commitedRepo);
  const [filtersSelected, setFiltersSelected] = useState<string[]>([]);

  useEffect(() => {
    if (filters.length) {
      setFiltersSelected([
        ...filtersSelected,
        ...filters.filter((filter) => filter.isActive && !filtersSelected.includes(filter.id)).map((filter) => filter.id)
      ]);
    }
  }, [filters]);

  const viewIssues = () => {
    commitOwner(owner);
    commitRepo(repo);
    setFilter(filtersSelected);
    navigation.goBack();
  };

  const handleFilterSelection = (id: string): void => {
    if (!filtersSelected.includes(id)) {
      setFiltersSelected([...filtersSelected, id]);
    } else {
      setFiltersSelected(filtersSelected.filter((item) => item !== id));
    }
  }

  const githubIcon = <AntDesign size={20} color={Colors.blue} name="github" style={{ marginRight: 6 }} />;

  return (
    <CustomView style={styles.container}>
      <View>
        <View style={styles.inputs}>
          <CustomTextInput name="Organization" testID="organizationInput" value={owner} onChangeText={setOwner} />
          <CustomText style={{ marginTop: 20, fontSize: 20, color: Colors.border }}>/</CustomText>
          <CustomTextInput name="Repository" value={repo} onChangeText={setRepo} />
        </View>
        <CustomText>Filters:</CustomText>
        {filters.map(({ id, label }) => (
          <TouchableOpacity
            onPress={() => handleFilterSelection(id)}
            activeOpacity={1}
            testID={id}
            key={id}
            style={[styles.filter, filtersSelected.includes(id) && styles.activeFilter]}
          >
            <CustomText>{label} issues</CustomText>
          </TouchableOpacity>
        ))}
      </View>
      <CustomButton style={styles.button} type="primary" text="View issues" testID="viewIssuesButton" leftIcon={githubIcon} onPress={viewIssues} />
    </CustomView>
  );
}

