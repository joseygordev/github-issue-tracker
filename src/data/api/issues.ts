import { GetIssuesParams, Issue } from '../../types/issues';
import methods from './methods';

export async function get(queryParams: GetIssuesParams): Promise<Issue[]> {
  const { owner, repo, filter, page } = queryParams;

  return await methods.get(`repos/${owner}/${repo}/issues?state=${filter}&page=${page}`);
}
