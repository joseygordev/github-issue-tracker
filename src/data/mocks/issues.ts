import { Issue } from '../../types/issues';
import mockUsers from './users';

export default [
  {
    id: 1,
    title: 'Mock',
    body: 'mock body',
    labels: [
      {
        id: '2',
        name: 'mock label',
        color: '#000',
        url: 'mock_url',
        default: false,
        description: 'mock description',
      },
    ],
    state: 'open',
    isBookmarked: false,
    user: mockUsers[0],
    comments_url: 'https://api.github.com/commentsUrl',
  } as Issue,
  {
    id: 2,
    title: 'Mock',
    body: 'mock body',
    labels: [
      {
        id: '2',
        name: 'mock label',
        color: '#000',
        url: 'mock_url',
        default: false,
        description: 'mock description',
      },
    ],
    state: 'closed',
    isBookmarked: true,
    user: mockUsers[1],
    comments_url: 'https://api.github.com/commentsUrl',
  } as Issue,
];

export const mockFilters = [
  { id: 'open', label: 'Open', isActive: true },
  { id: 'closed', label: 'Closed', isActive: true },
];