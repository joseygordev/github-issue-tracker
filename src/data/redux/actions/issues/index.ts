import { Dispatch } from 'redux';

import * as issueEndpoints from '../../../api/issues';

import { IssueActionType, IssueAction } from '../../../../types/issues';
import { AppState } from '../..';

export const getIssues = () => async (dispatch: Dispatch<IssueAction>, getState: () => AppState) => {
  dispatch({ type: IssueActionType.SET_ISSUES_LOADING });

  const { owner, repo, filters, currentPage } = getState().issuesReducer;

  let filter = filters.find((x) => x.isActive)?.id;
  if (!filter || filters.every((x) => x.isActive)) {
    filter = 'all';
  }

  try {
    const issues = await issueEndpoints.get({
      owner,
      repo,
      filter,
      page: currentPage,
    });

    dispatch({ type: IssueActionType.SET_ISSUES_SUCCESS, payload: issues });
  } catch (error) {
    dispatch({ type: IssueActionType.SET_ISSUES_ERROR, payload: error });
  }
};

export const commitOwner = (owner: string) => (dispatch: Dispatch<IssueAction>) => {
  dispatch({ type: IssueActionType.COMMIT_OWNER, payload: owner });
};

export const commitRepo = (repo: string) => (dispatch: Dispatch<IssueAction>) => {
  dispatch({ type: IssueActionType.COMMIT_REPO, payload: repo });
};

export const setPage = (pageNumber: number) => (dispatch: Dispatch<IssueAction>) => {
  dispatch({ type: IssueActionType.SET_PAGE, payload: pageNumber });
};

export const toggleFilter = (filterIds: string[]) => (dispatch: Dispatch<IssueAction>) => {
  dispatch({ type: IssueActionType.TOGGLE_FILTER, payload: filterIds });
};