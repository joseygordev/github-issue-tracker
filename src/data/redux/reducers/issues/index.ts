import { Issue, IssueActionType, IssueState, IssueAction } from '../../../../types/issues';

const defaultFilters = [
  { id: 'open', label: 'Open', isActive: true },
  { id: 'closed', label: 'Closed', isActive: true },
];

export const initialState: IssueState = {
  issues: [],
  owner: 'facebook',
  repo: 'react-native',
  filters: defaultFilters,
  loading: false,
  currentPage: 1,
};

export default (state: IssueState = initialState, action: IssueAction): IssueState => {
  switch (action.type) {
    case IssueActionType.SET_ISSUES_LOADING:
      return { ...state, loading: true };

    case IssueActionType.SET_ISSUES_SUCCESS:
      return {
        ...state,
        issues: action.payload as Issue[],
        loading: false,
        error: undefined,
      };

    case IssueActionType.SET_ISSUES_ERROR:
      return {
        ...state,
        loading: false,
        error: action.payload as Error,
        issues: [],
        currentPage: 1,
      };

    case IssueActionType.COMMIT_OWNER:
      return {
        ...state,
        owner: action.payload as string,
      };

    case IssueActionType.COMMIT_REPO:
      return {
        ...state,
        repo: action.payload as string,
      };

    case IssueActionType.SET_PAGE:
      return {
        ...state,
        currentPage: action.payload as number,
      };

    case IssueActionType.TOGGLE_FILTER:
      const payload = action.payload as string[];
      const newFilters = [...state.filters].map((filter) => ({...filter, isActive: !!payload.includes(filter.id)}));

      return {
        ...state,
        filters: newFilters,
        currentPage: 1,
      };

    default:
      return state;
  }
};
