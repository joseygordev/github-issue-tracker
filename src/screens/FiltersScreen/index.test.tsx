import React from 'react';
import { create, act } from 'react-test-renderer';
import { Provider as ReduxProvider } from 'react-redux';

import store from '../../data/redux';
import MockNavigation from '../../data/mocks/navigation';
import FiltersScreen from './index';
import { mockFilters } from '../../data/mocks/issues';
import CustomText from '../../components/CustomText';

const createFreshTree = () =>
  create(
    <ReduxProvider store={store}>
      <MockNavigation component={FiltersScreen} />
    </ReduxProvider>
  );

const tree = createFreshTree();

describe('Filters Screen', () => {
  it('renders correctly', async () => {
    await act(async () => jest.runAllTimers());

    expect(tree).toMatchSnapshot();
  });
  
  it('renders with owner set', async () => {
    const owner = tree.root.findByProps({ testID: 'ownerInput' }).props;

    expect(owner.value).toEqual('facebook');
  });

  it('renders with repo set', async () => {
    const owner = tree.root.findByProps({ testID: 'repoInput' }).props;

    expect(owner.value).toEqual('react-native');
  });
  
  it('renders with filters', async () => {
    const firstFilter = mockFilters[0];
    const filter = tree.root.findByProps({ testID: firstFilter.id }).props;

    expect(filter).toBeTruthy();
  });
});
