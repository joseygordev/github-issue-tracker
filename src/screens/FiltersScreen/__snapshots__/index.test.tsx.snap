// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Screen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  MockScreen
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#fff",
                            },
                            Object {
                              "flex": 1,
                              "flexDirection": "column",
                              "justifyContent": "space-between",
                              "padding": 16,
                            },
                          ]
                        }
                        testID="customView"
                      >
                        <View>
                          <View
                            style={
                              Object {
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                                "width": "100%",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "marginVertical": 12,
                                  "width": 335,
                                }
                              }
                            >
                              <View
                                pointerEvents="none"
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                    },
                                    Object {
                                      "zIndex": 1,
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#fff",
                                      "borderRadius": 8,
                                      "left": 0,
                                      "marginLeft": 6,
                                      "paddingHorizontal": 6,
                                      "position": "absolute",
                                      "top": -8,
                                      "zIndex": 20,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "color": "#000",
                                        },
                                        Object {
                                          "color": "#414d5b",
                                          "fontSize": 12,
                                        },
                                        Object {
                                          "fontFamily": "space-mono",
                                        },
                                      ]
                                    }
                                    testID="inputName"
                                  >
                                    Owner
                                  </Text>
                                </View>
                              </View>
                              <TextInput
                                autoCapitalize="none"
                                name="Owner"
                                onBlur={[Function]}
                                onChangeText={[Function]}
                                onFocus={[Function]}
                                placeholderTextColor="#aab2ba"
                                returnKeyType="done"
                                style={
                                  Array [
                                    Object {
                                      "borderRadius": 6,
                                      "borderWidth": 1,
                                      "height": 42,
                                      "paddingLeft": 13,
                                      "paddingRight": 25,
                                      "width": "100%",
                                    },
                                    Object {
                                      "borderColor": "#aab2ba",
                                    },
                                    undefined,
                                  ]
                                }
                                testID="ownerInput"
                                value="facebook"
                              />
                            </View>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#000",
                                  },
                                  Object {
                                    "color": "#aab2ba",
                                    "fontSize": 20,
                                    "marginTop": 20,
                                  },
                                  Object {
                                    "fontFamily": "space-mono",
                                  },
                                ]
                              }
                              testID="textValue"
                            >
                              /
                            </Text>
                            <View
                              style={
                                Object {
                                  "marginVertical": 12,
                                  "width": 335,
                                }
                              }
                            >
                              <View
                                pointerEvents="none"
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "flexDirection": "row",
                                    },
                                    Object {
                                      "zIndex": 1,
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "#fff",
                                      "borderRadius": 8,
                                      "left": 0,
                                      "marginLeft": 6,
                                      "paddingHorizontal": 6,
                                      "position": "absolute",
                                      "top": -8,
                                      "zIndex": 20,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "color": "#000",
                                        },
                                        Object {
                                          "color": "#414d5b",
                                          "fontSize": 12,
                                        },
                                        Object {
                                          "fontFamily": "space-mono",
                                        },
                                      ]
                                    }
                                    testID="inputName"
                                  >
                                    Repository
                                  </Text>
                                </View>
                              </View>
                              <TextInput
                                autoCapitalize="none"
                                name="Repository"
                                onBlur={[Function]}
                                onChangeText={[Function]}
                                onFocus={[Function]}
                                placeholderTextColor="#aab2ba"
                                returnKeyType="done"
                                style={
                                  Array [
                                    Object {
                                      "borderRadius": 6,
                                      "borderWidth": 1,
                                      "height": 42,
                                      "paddingLeft": 13,
                                      "paddingRight": 25,
                                      "width": "100%",
                                    },
                                    Object {
                                      "borderColor": "#aab2ba",
                                    },
                                    undefined,
                                  ]
                                }
                                testID="repoInput"
                                value="react-native"
                              />
                            </View>
                          </View>
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#000",
                                },
                                undefined,
                                Object {
                                  "fontFamily": "space-mono",
                                },
                              ]
                            }
                            testID="textValue"
                          >
                            Filters:
                          </Text>
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "borderColor": "#00aaff",
                                "borderRadius": 16,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "height": 42,
                                "marginVertical": 8,
                                "opacity": 1,
                                "paddingHorizontal": 16,
                                "paddingTop": 6,
                                "paddingVertical": 4,
                              }
                            }
                            testID="open"
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#000",
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "space-mono",
                                  },
                                ]
                              }
                              testID="textValue"
                            >
                              Open
                               issues
                            </Text>
                          </View>
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignItems": "center",
                                "borderColor": "#00aaff",
                                "borderRadius": 16,
                                "borderWidth": 1,
                                "flexDirection": "row",
                                "height": 42,
                                "marginVertical": 8,
                                "opacity": 1,
                                "paddingHorizontal": 16,
                                "paddingTop": 6,
                                "paddingVertical": 4,
                              }
                            }
                            testID="closed"
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#000",
                                  },
                                  undefined,
                                  Object {
                                    "fontFamily": "space-mono",
                                  },
                                ]
                              }
                              testID="textValue"
                            >
                              Closed
                               issues
                            </Text>
                          </View>
                        </View>
                        <View
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "center",
                              "backgroundColor": "transparent",
                              "borderColor": "#00aaff",
                              "borderRadius": 15,
                              "borderWidth": 1,
                              "flexDirection": "row",
                              "height": 42,
                              "justifyContent": "center",
                              "marginVertical": 8,
                              "minWidth": "100%",
                              "opacity": 1,
                              "width": 120,
                            }
                          }
                          testID="viewIssuesButton"
                        >
                          <Text
                            allowFontScaling={false}
                            selectable={false}
                            style={
                              Array [
                                Object {
                                  "color": "#00aaff",
                                  "fontSize": 20,
                                },
                                Object {
                                  "marginRight": 6,
                                },
                                Object {
                                  "fontFamily": "anticon",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }
                          >
                            
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "#000",
                                },
                                Array [
                                  Object {
                                    "color": "#00aaff",
                                    "fontSize": 12,
                                    "letterSpacing": 0.8,
                                  },
                                  Object {},
                                ],
                                Object {
                                  "fontFamily": "space-mono",
                                },
                              ]
                            }
                            testID="textValue"
                          >
                            VIEW ISSUES
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
